bot/exts/moderation/silence.py

```diff
@@ -102,61 +129,112 @@ async def unsilence(self, ctx: Context) -> None:
 
         If the channel was silenced indefinitely, notifications for the channel will stop.
         """
-        await self._get_instance_vars_event.wait()
+        await self._init_task
         log.debug(f"Unsilencing channel #{ctx.channel} from {ctx.author}'s command.")
-        if not await self._unsilence(ctx.channel):
-            await ctx.send(f"{Emojis.cross_mark} current channel was not silenced.")
+        await self._unsilence_wrapper(ctx.channel)
+
+    async def _unsilence_wrapper(self, channel: TextChannel) -> None:
+        """Unsilence `channel` and send a success/failure message."""
+        if not await self._unsilence(channel):
+            overwrite = channel.overwrites_for(self._verified_role)
+            if overwrite.send_messages is False or overwrite.add_reactions is False:
+                await channel.send(MSG_UNSILENCE_MANUAL)
+            else:
+                await channel.send(MSG_UNSILENCE_FAIL)
         else:
-            await ctx.send(f"{Emojis.check_mark} unsilenced current channel.")
+            await channel.send(MSG_UNSILENCE_SUCCESS)
 
-    async def _silence(self, channel: TextChannel, persistent: bool, duration: Optional[int]) -> bool:
-        """
-        Silence `channel` for `self._verified_role`.
+    async def _silence_overwrites(self, channel: TextChannel) -> bool:
```

Comment:
```suggestion
    async def _set_silence_overwrites(self, channel: TextChannel) -> bool:
```
To me, silence_overwrites either sounds like a variable that contains the overwrites for silences or a method that silences overwrites; not sure if adding the set or an another name would be a better choice

Reply:
See dc2797b